### ISSUE TITLE
SDL_HINT_VIDEO_FORCE_EGL and SDL_CreateWindowFrom() cannot be used together

### ISSUE THREAD

**Issue by Mangomm** [libsdl-org/SDL#8404]
base https://github.com/libsdl-org/SDL/issues/8302.

In the project I use an existing handle to create a window through SDL_CreateWindowFrom().
If you call SDL_SetHint(SDL_HINT_VIDEO_FORCE_EGL, "1") before creating a window; , it will report an error:
Couldn't find matching render driver.

Only comment out SDL_SetHint(SDL_HINT_VIDEO_FORCE_EGL, "1"); There will be no error.
I read the comments for SDL_HINT_VIDEO_FORCE_EGL, and if it is not mandatory to specify 1, SDL will use the platform-specific GL API(GLX, WGL, CGL, etc).

I actually tested it with 4K resolution, and when you create a window with the SDL_CreateWindowFrom() function and comment out the SDL_SetHint(SDL_HINT_VIDEO_FORCE_EGL, "1") statement, it still achieves high performance hard rendering.

But what if you want to use the SDL_CreateWindowFrom() function with SDL_HINT_VIDEO_FORCE_EGL? Can SDL be implemented?

I have already wrapped testffmpe.c in a Qt-based environment as a test example, which I can provide to you if you want to avoid repackaging.

**Comment by slouken** [comment]
Yes please, an example would be helpful, as well as more information about your system (OS, version, windowing system, etc.)

**Comment by Mangomm** [comment]
Linux OS：

admin1@admin1:~$ uname -a
Linux admin1 5.4.18-53-generic #42-KYLINOS SMP Fri Mar 4 03:34:54 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

qt version：

admin1@admin1:~$  qmake  -v
QMake version 3.1
Using Qt version 5.12.8 in /usr/lib/x86_64-linux-gnu

opengl version(by sdl code get version)：

  auto v = glGetString(GL_VERSION);
  if(v){
      SDL_Log("Opengl version %s\n", v);
  }

print:
INFO: Opengl version OpenGL ES 3.2 Mesa 20.0.8

demo(Inside I gave you two test videos in the testvideo directory)：
https://github.com/Mangomm/test-sdl-hwnd.

**Comment by Mangomm** [comment]
In fact, I think it should have nothing to do with the system version, the key point is:
When I called:
1. SDL_SetHint(SDL_HINT_VIDEO_FORCE_EGL, "1");  + SDL_CreateWindowAndRenderer() :

const char *extensions = eglQueryString(eglGetCurrentDisplay(), EGL_EXTENSIONS); What you get is:
EGL_ANDROID_blob_cache EGL_ANDROID_native_fence_sync EGL_CHROMIUM_sync_control EGL_EXT_buffer_age  EGL_EXT_create_context_robustness EGL_EXT_image_dma_buf_import EGL_EXT_image_dma_buf_import_modifiers  EGL_IMG_context_priority EGL_KHR_cl_event2 EGL_KHR_config_attribs EGL_KHR_create_context EGL_KHR_create_context_no_error  EGL_KHR_fence_sync EGL_KHR_get_all_proc_addresses EGL_KHR_gl_colorspace EGL_KHR_gl_renderbuffer_image  EGL_KHR_gl_texture_2D_image EGL_KHR_gl_texture_3D_image EGL_KHR_gl_texture_cubemap_image EGL_KHR_image  EGL_KHR_image_base EGL_KHR_image_pixmap EGL_KHR_no_config_context EGL_KHR_reusable_sync EGL_KHR_surfaceless_context  EGL_EXT_pixel_format_float EGL_KHR_wait_sync EGL_MESA_configless_context EGL_MESA_drm_image  EGL_MESA_image_dma_buf_export EGL_MESA_query_driver EGL_NOK_texture_from_pixmap EGL_WL_bind_wayland_display

It contains the "EGL_EXT_image_dma_buf_import" substring, so has_EGL_EXT_image_dma_buf_import = SDL_TRUE;

But when I call it like this:
2. Comment out SDL_SetHint(SDL_HINT_VIDEO_FORCE_EGL, "1"), then call p->window = SDL_CreateWindowFrom(p->hwnd);

const char *extensions = eglQueryString(eglGetCurrentDisplay(), EGL_EXTENSIONS); What you get is:
EGL_EXT_platform_base EGL_EXT_device_base EGL_EXT_device_enumeration EGL_EXT_device_query  EGL_KHR_client_get_all_proc_addresses EGL_EXT_client_extensions EGL_KHR_debug EGL_KHR_platform_x11 EGL_EXT_platform_x11  EGL_EXT_platform_device EGL_KHR_platform_gbm EGL_MESA_platform_gbm EGL_EXT_platform_wayland  EGL_MESA_platform_surfaceless

It does not contain the "EGL_EXT_image_dma_buf_import" substring, so has_EGL_EXT_image_dma_buf_import = SDL_FALSE;

The above situation, you can see in https://github.com/Mangomm/test-sdl-hwnd testffmpeg.cpp,  control #define USING_HANDLE to reproduce.

**Comment by Mangomm** [comment]
Hello, is there a solution to this problem? Is there any progress?

**Comment by slouken** [comment]
I don't think anyone has had a chance to look yet.

**Comment by slouken** [comment]
We are scoping work for the SDL 3.2.0 release, so please let us know if this is a showstopper for you.

**Comment by Raffaello** [comment]
@Mangomm **I suppose it's happening because** the number is the index of the EGL device driver to use, so your error means there is no device driver with that index = 1

This is most likely because you have just 1  (or zero) egl device, as in C are zero based you should put 0  instead of 1 and 0 is equivalent to disable to: in toher words it is pointless to use this hint with only 1 egl device.

soruce code reference from SDL_egl.c (branch release-2.30.x)
```cpp
 if (device >= num_egl_devices) {
            return SDL_SetError("Invalid EGL device is requested.");
        }
```
